packages/plugin-list/src/ListXmlDomParser.ts

```diff
@@ -15,17 +15,52 @@ export class ListXmlDomParser extends AbstractParser<Node> {
     };
 
     /**
-     * Parse a list (UL, OL) and his children list elements (LI).
+     * Parse a list (UL, OL) and its children list elements (LI).
      *
      * @param context
      */
     async parse(item: Element): Promise<VNode[]> {
-        if (!listTags.includes(nodeName(item))) return;
-        const listType = nodeName(item) === 'UL' ? ListType.UNORDERED : ListType.ORDERED;
-        const listNode = new ListNode({ listType });
-        listNode.attributes = this.engine.parseAttributes(item);
-        const nodes = await this.engine.parse(...item.childNodes);
-        listNode.append(...nodes);
-        return [listNode];
+        // Get the list's type (ORDERED, UNORDERED, CHECKLIST).
+        let type: ListType;
+        if (item.className.match(/(^| )checklist( |$)/i)) {
+            type = ListType.CHECKLIST;
+        } else {
+            type = nodeName(item) === 'UL' ? ListType.UNORDERED : ListType.ORDERED;
+        }
+
+        // Create the list node and parse its children and attributes.
+        const list = new ListNode({ listType: type });
+        const children = await this.engine.parse(...item.childNodes);
+        list.append(...children);
+        list.attributes = this.engine.parseAttributes(item);
+
+        // In the case of a checklist, parse their checked/unchecked status and
+        // ensure vertical propagatation.
+        if (type === ListType.CHECKLIST) {
+            const reChecked = /(^| )checked( |$)/i;
+            for (const child of children) {
+                const liAttributes = child.attributes['li-attributes'] as Record<string, string>;
+                if (!liAttributes?.class) continue;
```

Comment:
I *hate* `continue`s and `break`s when they can be written as proper `filter` or `while` conditions because the latter is always more idiomatic than the former.

In this case:
```typescript
const items = children.filter(child => child.modifiers.find(LiAttributes)?.class);
``` 
If you can somehow remove the `const next = child.nextSibling();` below then you could even iterate over the `LiAttributes` rather than the nodes themselves. (Especially considering I don't understand the case for which this part of the code exist, I don't understand why the last checked item must be treated differently.)

Reply:
I need `classList` but I also need `child` so I can check it.
Therefore using filter means I'm going to have to go find the attributes for each child twice:
```ts
const items = children.filter(child => child.modifiers.find(ListItemAttributes));
for (const item of items) {
    const liAttributes = item.modifiers.find(ListItemAttributes);

    // Parse the list item's checked status.
    if (liAttributes.classList.has('checked')) {
        ListNode.check(item);
    }

    // Remove the checklist-related classes from `liAttributes`.
    liAttributes.classList.remove('checklist checked unchecked');
}
```
I don't think we want that. Right now I'm putting an `if` instead.